src/components/Pressable/Pressable.tsx

```diff
@@ -254,8 +257,13 @@ export default function Pressable(props: PressableProps) {
       ? props.children({ pressed: pressedState })
       : props.children;
 
+  const flattenedStyles = StyleSheet.flatten((styleProp ?? {}) as ViewStyle);
+
+  const innerStyles = extractStyles(flattenedStyles, innerStyleKeys);
+  const outerStyles = excludeStyles(flattenedStyles, innerStyleKeys);
```

Comment:
Since styles are getting divided into two exclusive groups, wouldn't it be faster to implement one method that would divide them in a single pass? Something like this:

```jsx
const [innerStyles, outerStyles] = dividePressableStyle(flattenedStyles)
```

For now, since it's not used anywhere else, I'd keep `innerStyleKeys` internally in utils and reference directly.

WDYT?

Reply:
Yep, I combined both methods in [5595026](https://github.com/software-mansion/react-native-gesture-handler/pull/2982/commits/55950265f13d9d3bff235719fc5889dd20b219c8), but I'm still yet to optimize it to create both parts in a single pass instead of doing it in 2 passes.

Reply:
As of [e348d8f](https://github.com/software-mansion/react-native-gesture-handler/pull/2982/commits/e348d8fcd1eacb14e9c36cc4bd7e51ad44e602ea), both `inner` and `outer` style separation is combined into a single `forEach` statement.